cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -348,7 +351,15 @@ func BuildAWS(opts config.AutoscalingOptions, do cloudprovider.NodeGroupDiscover
 		klog.Fatalf("Failed to create AWS Manager: %v", err)
 	}
 
-	provider, err := BuildAwsCloudProvider(manager, rl)
+	// Generate EC2 list
```

Comment:
I do not like the idea to fallback to static list of instances if runtime detection does not work. With this we can end up with  unpredicatable behavior of AWS cp if there is some random network/API outage while Cluster-Autoscaler is starting up.
I would suggest adding an program flag to specify if static or dynamic list should be used. And if dynamic is used and API endpoint cannot be queried, return an error from BuildAWS method.

Reply:
@Jeffwan Are you still working on this one?

Reply:
Yes. I update the PR and expose an option. Please have a check